VAMobile/src/components/Templates/HeaderBanner.test.tsx

```diff
@@ -40,74 +33,54 @@ context('HeaderBanner', () => {
   })
   describe('left banner button', () => {
-    it('should not be there', async () => {
-      const leftButton = testInstance.findAllByType(TouchableWithoutFeedback)
-      expect(leftButton.length).toEqual(0)
-    })
-    it('should be there when text and on press is added', async () => {
+    it('should be there when text and on press is added', () => {
       initializeTestInstance(undefined, 'cancel', onPressSpy)
-      const leftButton = testInstance.findAllByType(TouchableWithoutFeedback)
-      expect(leftButton.length).toEqual(1)
-      const textViews = testInstance.findAllByType(TextView)
-      expect(textViews.length).toEqual(1)
-      expect(textViews[0].props.children).toEqual('cancel')
+      expect(screen.getByRole('button', { name: 'cancel' })).toBeTruthy()
+      fireEvent.press(screen.getByRole('button', { name: 'cancel' }))
+      expect(onPressSpy).toHaveBeenCalled()
     })
   })
 
   describe('right banner button', () => {
-    it('should not be there', async () => {
-      const rightButton = testInstance.findAllByType(TouchableWithoutFeedback)
-      expect(rightButton.length).toEqual(0)
-    })
-    it('should be there when text and onpress is added', async () => {
+    it('should be there when text and onpress is added', () => {
       initializeTestInstance(undefined, undefined, undefined, 'done', onPressSpy)
-      const rightButton = testInstance.findAllByType(TouchableWithoutFeedback)
-      expect(rightButton.length).toEqual(1)
-      const textViews = testInstance.findAllByType(TextView)
-      expect(textViews.length).toEqual(1)
-      expect(textViews[0].props.children).toEqual('done')
+      expect(screen.getByRole('button', { name: 'done' })).toBeTruthy()
+      fireEvent.press(screen.getByRole('button', { name: 'done' }))
+      expect(onPressSpy).toHaveBeenCalled()
     })
     it('should not have an icon when only text and on press is supplied', async () => {
       initializeTestInstance(undefined, undefined, undefined, 'done', onPressSpy)
-      const icon = testInstance.findAllByType(VAIcon)
-      expect(icon.length).toEqual(0)
+      expect(screen.UNSAFE_queryAllByType(VAIcon).length).toEqual(0)
     })
-    it('should have an icon w/ text button when text, onpress, and icon props are supplied', async () => {
+    it('should have an icon w/ text button when text, onpress, and icon props are supplied', () => {
       const rightIconProps: VAIconProps = {
         name: 'ProfileSelected',
         fill: 'largeNav',
         height: 22,
         width: 22,
       }
-      initializeTestInstance(undefined, undefined, undefined, 'done', onPressSpy, rightIconProps)
-      const icon = testInstance.findAllByType(VAIcon)
-      expect(icon.length).toEqual(1)
+      initializeTestInstance(undefined, undefined, undefined, 'icon button', onPressSpy, rightIconProps)
+      expect(screen.getByRole('button', { name: 'icon button' })).toBeTruthy()
+      expect(screen.UNSAFE_getByType(VAIcon)).toBeTruthy()
```

Comment:
there should be another way to test for this instead of the unsafe check, but it also may not even be needed honestly

Reply:
removed all VAIcon tests